pandas/core/indexes/base.py

```diff
@@ -5866,7 +5866,7 @@ def drop(self: _IndexT, labels, errors: str_t = "raise") -> _IndexT:
         mask = indexer == -1
         if mask.any():
             if errors != "ignore":
-                raise KeyError(f"{labels[mask]} not found in axis")
+                raise KeyError(f"{labels[mask]} not found in columns")
```

Comment:
You can not do it this way. We are running through here too, if the user drops something from the index. This would be very misleading. Also you have to adjust tests

Reply:
Sure, so what do you think will be the right way to do ?

Reply:
@phofl Does `Index` have an attribute that determines whether it is a column or index axis? If it does, it could use that the change the error message. If not, an alternative is to catch this error in `DataFrame.drop` and rethrow with a more specific message.